ilib-scanner.js

```diff
@@ -101,6 +145,26 @@ function loadIlibClasses() {
     }
 }
 
+function maplocaleCalDate(locales){
+    locales.forEach(function(lo){
+        var locale = new Locale(lo);
+        var region = locale.getRegion();
+        for (var item in localeCalMap){
+            if (item === region) {
+                var isValueString = (typeof localeCalMap[item] === "string")? true : false;
+                var caldate = (isValueString) ? localeCalMap[item] : localeCalMap[item][0];
+                if (!addCalDateList.includes(caldate)){
```

Comment:
This will fail to add CopticDate if your locale list is ["tr-TR","ar-EG"]

Maybe it is better to implement this as a Set object. Add all calendar dates to the set for every region, and then in the end, ask the set for an array of its values. That will take care of the duplicates automatically.

First, convert all values in localeCalMap to arrays. (Most are 1 element arrays). Then:

```javascript
var dateTypes = new Set();
locales.forEach(function(lo){
  var locale = new Locale(lo);
  var region = locale.getRegion();
  var dates = localeCalMap[region];
  if (dates) {
    dates.forEach(function(type) {
      dateTypes.add(type);
    });
  }
});
addCalDateList = toArray(dateTypes);

Reply:
You'll have to implement `toArray()` of course

Reply:
Yes,I agree to use Set instead of Array. I updated codes and became simpler.  I verified the output generate correctly. Please review it again.